torch/testing/_internal/jit_metaprogramming_utils.py

```diff
@@ -57,9 +56,9 @@
     ('max_pool2d', (S, S, S, S), (2, 1), '', (True, 'aten::max_pool2d_with_indices')),
     ('max_pool2d', (S, S, S, S), (2, 1, 1, 1, False, True), 'with_indices', (True, 'aten::max_pool2d_with_indices')),
     ('max_pool3d', (S, S, S, S, S), (2, 1)),
-    ('max_unpool1d', torch.tensor([[[2., 4]]]), (torch.tensor([[[1, 3]]]), 2, 2, 0)),
-    ('max_unpool2d', torch.tensor([[[[2., 4]]]]), (torch.tensor([[[[1, 3]]]]), 2, 2, 0)),
-    ('max_unpool3d', torch.tensor([[[[[2., 4]]]]]), (torch.tensor([[[[[1, 3]]]]]), 2, 2, 0)),
+    ('max_unpool1d', torch.tensor([[[2., 4]]], dtype=default_dtype), (torch.tensor([[[1, 3]]]), 2, 2, 0)),
```

Comment:
Whoa, I had no idea this list existed. @kurtamohler are these tests redundant with our OpInfo-based jit tests?

cc @suo @eellison 

If we can remove this list that'd be awesome. Seems like it must be mergeable with our OpInfo-based tests?

Reply:
In the long-term, yes, I would like to remove them. In the short term, it's tough, I think we need to separate out all of the jit different testing variations into separate tests. As is, it's pretty common for one test case to disable testing for all downstream tests. For example, Conv empty input breaks our alias analysis test, so now all jit tests are disabled for conv. Id be worried about getting rid of these tests which are pretty critical for autodiff testing because the OpInfo tests so frequently are disabled. 

Opinfos have a lot of other, much better properties but I feel more safe having these tests around in the short term

Reply:
Would you file an issue to track this refactoring and your concerns/thoughts about how best to do it, then @eellison? We can apply the "better engineering" label to it.